src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/sessiondays/SessionDaysReport.java

```diff
@@ -0,0 +1,271 @@
+            if (CollectionUtils.isNotEmpty(dates)) {
+                for (DateListedTypeItem d : dates) {
+                    if (isHearingStatusValid(d)) {
+                        String judgeStatus = getJudgeStatus(h.getValue().getJudge());
+                        SessionDaysReportSummary2 reportSummary2 = getReportSummary2Item(
+                                d.getValue(), sessionDaysReportSummary2List);
+                        switch (judgeStatus) {
+                            case "SALARIED":
+                                ft = Integer.parseInt(reportSummary.getFtSessionDaysTotal()) + 1;
+                                reportSummary.setFtSessionDaysTotal(String.valueOf(ft));
+                                ft2 = Integer.parseInt(reportSummary2.getFtSessionDays()) + 1;
+                                reportSummary2.setFtSessionDays(String.valueOf(ft2));
+                                break;
+                            case "FEE_PAID":
+                                pt = Integer.parseInt(reportSummary.getPtSessionDaysTotal()) + 1;
+                                reportSummary.setPtSessionDaysTotal(String.valueOf(pt));
+                                pt2 = Integer.parseInt(reportSummary2.getPtSessionDays()) + 1;
+                                reportSummary2.setPtSessionDays(String.valueOf(pt2));
+                                break;
+                            case "":
+                                ot = Integer.parseInt(reportSummary.getOtherSessionDaysTotal()) + 1;
+                                reportSummary.setOtherSessionDaysTotal(String.valueOf(ot));
+                                ot2 = Integer.parseInt(reportSummary2.getOtherSessionDays()) + 1;
+                                reportSummary2.setOtherSessionDays(String.valueOf(ot2));
+                                break;
+                            default:
+                                break;
+                        }
+                        int total = ft2 + pt2 + ot2;
+                        reportSummary2.setSessionDaysTotalDetail(String.valueOf(total));
+                    }
+                }
+            }
+        }
+    }
+
+    private boolean isHearingStatusValid(DateListedTypeItem d) {
+        return HEARING_STATUS_HEARD.equals(d.getValue().getHearingStatus());
+    }
+
+    private String getJudgeStatus(String judgeName) {
+        Judge j = jpaJudgeService.getJudge(office, judgeName);
+        if (j != null) {
+            return j.getEmploymentStatus().name();
+        }
+        return "";
+    }
+
+    private List<HearingTypeItem> getHearings(SessionDaysCaseData caseData) {
+        var hearings = caseData.getHearingCollection();
+        if (hearings == null) {
+            return Collections.emptyList();
+        }
+        return hearings;
+    }
+
+    private void setReportDetail(SessionDaysCaseData caseData, List<SessionDaysReportDetail> reportDetailList) {
+        for (HearingTypeItem h : getHearings(caseData)) {
```

Comment:
personally, would change the name of 'h' into something more meaningful, maybe refactor to hearingTypeItem? same with 'd' below

Reply:
refactored